drivers/arm_scmi/shell.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * System Control and Management Interface (SCMI) Shell interface
+ *
+ * Copyright (c) 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/firmware/arm_scmi.h>
+#include <zephyr/shell/shell.h>
+
+#include "base.h"
+
+#if defined(CONFIG_DT_HAS_ARM_SCMI_SMC_ENABLED)
+static const struct device *scmi_dev = DEVICE_DT_GET_ANY(arm_scmi_smc);
+#else
+BUILD_ASSERT(1, "unsupported scmi interface");
```

Comment:
I believe you can do this in KConfig. Something like

`depends on DT_HAS_ARM_SCMI_SMC_ENABLED`

Reply:
Here no  as diferent compatible need to be selected